Fix OrderWatcher uncaught Error

prettymuchbryce opened the pull request:
## Description

Addresses #761 

## Testing instructions

I tested these changes locally by logging out the errors that occurred. I couldn't figure out a way to write a graceful unit test, but I am open to suggestions on how to write one.

e.g.

```typescript
        let decodedData: any[];
        try {
          decodedData = ethersInterface.events[event.name].parse(log.data);
        } catch (error) {
          console.log('got error', error);
          if (error.code === ethers.errors.INVALID_ARGUMENT) {
            console.log('was invalid_argument');
            return log;
          }
          throw error;
        }
```

```
got error { Error: insufficient data for uint256 type (arg="_value", coderType="uint256", value="0x")
    at Object.throwError (/Users/bryce/projects/0x-monorepo/node_modules/ethers/utils/errors.js:76:17)
    at Object.decode (/Users/bryce/projects/0x-monorepo/node_modules/ethers/utils/abi-coder.js:83:24)
    at /Users/bryce/projects/0x-monorepo/node_modules/ethers/utils/abi-coder.js:386:32
    at Array.forEach (<anonymous>)
    at unpack (/Users/bryce/projects/0x-monorepo/node_modules/ethers/utils/abi-coder.js:379:12)
    at Object.decode (/Users/bryce/projects/0x-monorepo/node_modules/ethers/utils/abi-coder.js:524:26)
    at Coder.decode (/Users/bryce/projects/0x-monorepo/node_modules/ethers/utils/abi-coder.js:689:48)
    at EventDescription.result.parse (/Users/bryce/projects/0x-monorepo/node_modules/ethers/contracts/interface.js:327:60)
    at AbiDecoder.tryToDecodeLogOrNoop (/Users/bryce/projects/0x-monorepo/packages/utils/lib/abi_decoder.js:32:62)
    at OrderWatcher.<anonymous> (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/order_watcher.js:228:76)
    at step (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/order_watcher.js:32:23)
    at Object.next (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/order_watcher.js:13:53)
    at /Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/order_watcher.js:7:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/order_watcher.js:3:12)
    at OrderWatcher._onEventWatcherCallbackAsync (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/order_watcher.js:214:16)
    at EventWatcher.<anonymous> (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/event_watcher.js:147:25)
    at step (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/event_watcher.js:40:23)
    at Object.next (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/event_watcher.js:21:53)
    at /Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/event_watcher.js:15:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/event_watcher.js:11:12)
    at EventWatcher._emitDifferencesAsync (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/event_watcher.js:140:16)
    at EventWatcher.<anonymous> (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/event_watcher.js:109:51)
    at step (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/event_watcher.js:40:23)
    at Object.next (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/event_watcher.js:21:53)
    at fulfilled (/Users/bryce/projects/0x-monorepo/packages/order-watcher/lib/src/order_watcher/event_watcher.js:12:58)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
  reason: 'insufficient data for uint256 type',
  code: 'INVALID_ARGUMENT',
  arg: '_value',
  coderType: 'uint256',
  value: '0x' }
was invalid_argument
```

## Types of changes

*   [X] Bug fix (non-breaking change which fixes an issue)
*   [ ] New feature (non-breaking change which adds functionality)
*   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [ ] Change requires a change to the documentation.
*   [ ] Added tests to cover my changes.
*   [X] Added new entries to the relevant CHANGELOG.jsons.
*   [ ] Labeled this PR with the 'WIP' label if it is a work in progress.
*   [ ] Labeled this PR with the labels corresponding to the changed package.

fabioberger commented on packages/order-watcher/CHANGELOG.json:
```diff
@@ -1,4 +1,14 @@
 [
+    {
+        "timestamp": 1529701469,
```
Please omit the timestamp. This get's added auto-magically at publish time.

fabioberger commented on packages/order-watcher/src/order_watcher/order_watcher.ts:
```diff
@@ -241,7 +242,15 @@ export class OrderWatcher {
+        try {
+            maybeDecodedLog = this._web3Wrapper.abiDecoder.tryToDecodeLogOrNoop<ContractEventArgs>(log);
+        } catch (error) {
+          if (error.code === ethers.errors.INVALID_ARGUMENT) {
```
Can you add a `// HACK: Some description` comment above this explaining the rationale for ignoring these errors for the next engineer who chances upon this?

prettymuchbryce commented:
@fabioberger I have addressed your feedback. Also I have decided to move the try/catch down the stack into the AbiDecoder instead as it seems like a more fitting place to handle this error.

prettymuchbryce commented:
Also I'm not really sure how you typically manage version bumps / yarn.lock updates. I have bumped all of the versions here in the `package.json`, but please let me know if there's anything else I need to do.

fabioberger commented on packages/typescript-typings/types/ethers/index.d.ts:
```diff
@@ -31,4 +31,7 @@ declare module 'ethers' {
         public static getDeployTransaction(bytecode: string, abi: any, ...args: any[]): Partial<TxData>;
         constructor(address: string, abi: any, provider: any);
     }
+    const enum errors {
```
Convention: Enum variable name should start with capital letter.

prettymuchbryce replied:
This is a definition file for an existing library (ethers). My experience with TypeScript is limited, but my understanding is that I'm just defining the type for this existing member and I cannot change the name or the casing.

fabioberger replied:
Ah yep, that is correct!

fabioberger commented:
Thanks for the fix @prettymuchbryce, is this blocking you or can we wait with publishing a new version?

prettymuchbryce commented:
Thanks @fabioberger. Please publish at your discretion. It is not blocking me.
